llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -2960,6 +2967,197 @@ static bool validateAndCostRequiredSelects(BasicBlock *BB, BasicBlock *ThenBB,
   return HaveRewritablePHIs;
 }
 
+static bool isLoadFromAlloca(const Instruction &I) {
+  return isa<LoadInst>(I) && isa<AllocaInst>(I.getOperand(0));
```

Comment:
Maybe:
`isa<LoadInst>(I) && isa<AllocaInst>(getUnderlyingObject(I.getOperand(0))`

Reply:
I see now that this will also req checks that all intermediate operands dominate the br

Reply:
Personally would drop this helper and make it a lambda in the below function, it doesn't seem general.

Reply:
Good point! Will do.

Reply:
Done.